Agda.cabal

```diff
@@ -569,6 +569,7 @@ library
       Agda.Syntax.Scope.Flat
       Agda.Syntax.Scope.Monad
       Agda.Syntax.TopLevelModuleName
+      Agda.Syntax.TopLevelModuleName.Boot
```

Comment:
Naming-wise, I think we already have some modules named `Something.Base`, so perhaps we could keep the same naming convention here.

Reply:
What if we need to split a module that already ends with `.Base`?

e.g. when splitting `Agda.TypeCheckecking.Monad.Base`, what should the new module be called?

* `Agda.TypeChecking.Monad.Base.Base`
* `Agda.TypeChecking.Monad.Base.Boot`
* `Agda.TypeChecking.Monad.Boot`
* `Agda.TypeChecking.Monad.Base2`

Reply:
For the moment I am using `Boot` because it is fresh -- so it's easier to grep for if we want to change it later.

Reply:
I think `.Boot` indicates that the sole purpose of the module is to break up a dependency cycle.
`.Base` is more generic.  (In other Haskell projects, there is also a convention to use `.Types` for just collecting the data type definitions pertaining to a particular purpose.)

In this case, I would think that `.Boot` is appropriate, but `.Base` is thinkable because it isn't yet taken, so I don't really have a strong opinion here.  When we split `.Base` modules, we might have to introduce `.Boot` modules, so we could already do it here...